services/provider/server/server.go

```diff
@@ -105,23 +108,47 @@ func NewOCSProviderServer(ctx context.Context, namespace string) (*OCSProviderSe
 		return nil, fmt.Errorf("failed to create new scheme. %v", err)
 	}
 
-	client, err := util.NewK8sClient(scheme)
+	config, err := config.GetConfig()
 	if err != nil {
-		return nil, fmt.Errorf("failed to create new client. %v", err)
+		return nil, err
+	}
+
+	mgr, err := manager.New(config, manager.Options{
+		Scheme: scheme,
+		Cache: cache.Options{
+			DefaultNamespaces: map[string]cache.Config{
+				namespace: {},
+			},
+		},
+	})
```

Comment:
Can you please why do we need to use a controller manager in the server where the only thing we want is a cached backed client?